### PR TITLE
Fix: create a config with the same name reported an incorrect error

### LINKED ISSUE #5239: [Bug] create a config with the same name reported an incorrect error
**Describe the bug**

create a config with the same name reported an incorrect error, the same name check did not work, and method CreateOrUpdateConfig does not handle creation and update separately, so it needs to do the same name check during creation, but not for update
image
image

**To Reproduce**

**Expected behavior**

**Screenshots**

**KubeVela Version**
1.6.5

**Cluster information**

**Additional context**

### PR DESCRIPTION
**Pull request by nuclearwu**
Signed-off-by: wuzhongjian <wuzhongjian_yewu@cmss.chinamobile.com>

### Description of your changes

Fixes #5239 

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/apiserver/utils/bcode/016_config.go [hunk 1]
```diff
@@ -37,4 +37,7 @@ var (
 
 	// ErrNotFoundDistribution means the distribution is not exist
 	ErrNotFoundDistribution = NewBcode(404, 16007, "the distribution is not exist")
+
+	// ErrChangeAuthSwitch the template of the config can not be change
+	ErrChangeAuthSwitch = NewBcode(400, 16008, "the auth switch of the config can not be change")
```
Comment (barnettZQG) [inline pkg/apiserver/utils/bcode/016_config.go:42]: The error message is not clear.

Comment (nuclearwu) [reply pkg/apiserver/utils/bcode/016_config.go]: @barnettZQG It has been changed to "the auth switch of the image registry config can not be change", please review

**Comment by nuclearwu** [comment]
@barnettZQG

File: pkg/apiserver/utils/bcode/016_config.go [hunk 2]
```diff
@@ -37,4 +37,7 @@ var (
 
 	// ErrNotFoundDistribution means the distribution is not exist
 	ErrNotFoundDistribution = NewBcode(404, 16007, "the distribution is not exist")
+
+	// ErrChangeAuthSwitch the template of the config can not be change
+	ErrChangeAuthSwitch = NewBcode(400, 16008, "the auth switch of the image registry config can not be change")
```
Comment (barnettZQG) [inline pkg/apiserver/utils/bcode/016_config.go:42]: Not only the image registry config template could change the secret type. So, this error type and description should have more universality.

Comment (nuclearwu) [reply pkg/apiserver/utils/bcode/016_config.go]: @barnettZQG ok, have done, please review

**Review by barnettZQG — APPROVED** [review]
lgtm

**Pull request merged**